src/SearchParameters/index.js

```diff
@@ -403,15 +410,29 @@ SearchParameters.prototype = {
     } );
   },
   /**
-   * Remove a refinement set on facet.
+   * Remove a refinement set on facet. If a value is provided, it will clear the
+   * refinement for the given value, otherwise it will clear all the refinement
+   * values for the facetted attribute.
    * @method
    * @param {string} facet
+   * @param {string} value
    * @return {SearchParameters}
    */
-  removeFacetRefinement : function removeFacetRefinement( facet ) {
+  removeFacetRefinement : function removeFacetRefinement( facet, value ) {
     return this.mutateMe( function( m ) {
       m.page = 0;
-      m._clearFacetRefinements( facet );
+      if( value ) {
+        var idx = m.facetsRefinements[ facet ].indexOf( value );
+        if( idx > -1 ) {
+          m.facetsRefinements[ facet ].splice( idx, 1 );
+          if( m.facetsRefinements[ facet ].length === 0 ) {
+            delete m.facetsRefinements[ facet ];
```

Comment:
actually we remove the key from the object so when iterating with a `for in` we don't need to check for the content.